plugins/org.csstudio.trayicon/src/org/csstudio/trayicon/TrayApplicationWorkbenchWindowAdvisor.java

```diff
@@ -89,23 +89,26 @@ public boolean preWindowShellClose() {
 
         boolean closeWindow;
         int userAction = DIALOG_CLOSED;
+        int numWorkspaces = PlatformUI.getWorkbench().getWorkbenchWindowCount();
```

Comment:
I don't think `numWorkspaces` is a good name here.  Workspace is the directory for saving preferences.  `numWindows`?